File: EconRentCarWeb/EconRentCar.DataModel/TipoVehiculo.cs
using EconRentCar.Core;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EconRentCar.DataModel
{
    public class TipoVehiculo : IEntityBase
    {
        public TipoVehiculo()
        {
            Vehiculos = new HashSet<Vehiculo>();
        }
        public Guid Id { get; set; }
        [Required]
        public string Nombre { get; set; }
        [Required]
        public string Descripcion { get; set; }
        public bool Activo { get; set; }
        public virtual ICollection<Vehiculo> Vehiculos { get; set; }
    }
}

File: EconRentCarWeb/EconRentCarClient/src/app/shared/models/tipo-vehiculo.ts
import { BaseData } from '../dynamic-crud/models';
import { Vehiculo } from './vehiculo';

export class TipoVehiculo implements BaseData {
    public Id = '00000000-0000-0000-0000-000000000000';
    public Nombre = '';
    public Descripcion = '';
    public Activo = false;
    public Vehiculos: Vehiculo[];
}
